src/VirtoCommerce.Platform.Web/Security/ServiceCollectionExtensions.cs

```diff
@@ -47,5 +51,22 @@ public static IServiceCollection AddSecurityServices(this IServiceCollection ser
 
             return services;
         }
+
+        public static void AddPlatformDataProtection(this IServiceCollection services, IConfiguration config)
+        {
+            var redisConnectionString = config.GetConnectionString("RedisConnectionString");
+            if (!string.IsNullOrEmpty(redisConnectionString))
+            {
+                var redis = ConnectionMultiplexer.Connect(redisConnectionString);
+                services.AddDataProtection()
+                    .SetApplicationName("VirtoCommerce.Platform")
+                    .PersistKeysToStackExchangeRedis(redis, "VirtoCommerce-Keys");
+            }
+            else
+            {
+                var dataProtectionKeysFolder = new DirectoryInfo(Path.GetFullPath(Path.Combine("app_data","dataprotectionkeys")));
```

Comment:
This section won't work for multiple instances because it requires a shared folder.